src/hypercall.rs

```diff
@@ -84,13 +85,49 @@ pub fn unlink(mem: &MmapMemory, sysunlink: &mut UnlinkParams) {
 }
 
 /// Handles an open syscall by opening a file on the host.
-pub fn open(mem: &MmapMemory, sysopen: &mut OpenParams) {
-	unsafe {
-		sysopen.ret = libc::open(
-			mem.host_address(sysopen.name).unwrap() as *const i8,
-			sysopen.flags,
-			sysopen.mode,
-		);
+pub fn open(mem: &MmapMemory, sysopen: &mut OpenParams, file_map: &Option<UhyveFileMap>) {
+	// TODO: We could keep track of the file descriptors internally, in case the kernel doesn't close them.
+	let requested_path = mem.host_address(sysopen.name).unwrap() as *const i8;
+
+	// If the file_map doesn't exist, the provided path will be used instead.
+	// (i.e. host filesystem access).
+	if let Some(file_map) = file_map {
+		// Rust deals in UTF-8. C doesn't provide such a guarantee.
+		// In that case, converting a CStr to str will return a Utf8Error.
+		//
+		// See: https://nrc.github.io/big-book-ffi/reference/strings.html
+		let guest_path = unsafe { CStr::from_ptr(requested_path) }.to_str();
+
+		if let Ok(guest_path) = guest_path {
+			let paths = file_map;
+			let host_path_option = paths.get_paths().get_key_value(guest_path);
+
+			if let Some((_guest_path, host_path)) = host_path_option {
```

Comment:
Merging these lines would be a bit more "Rusty"

Reply:
Also, why are you using `get_key_value` if the key is not used afterwards? `get` is what you want to use here :slightly_smiling_face:

Reply:
I intended to use the key somehow, but it seems like I changed my mind. Addressed.

https://github.com/hermit-os/uhyve/commit/fc325cd5b7557539bc2d6eac14fb41dea7add6ec